SSA/Projects/FullyHomomorphicEncryption/Basic.lean

```diff
@@ -43,6 +48,62 @@ theorem WithBot.npow_coe_eq_npow (n : Nat) (x : ℕ) : (WithBot.some x : WithBot
 
 noncomputable def f : (ZMod q)[X] := X^(2^n) + 1
 
+-- theorem zmodq_eq_finq : ZMod q = Fin q := by
+--   have h : q > 1 := hqgt1.elim
+--   unfold ZMod
+--   cases q
+--   · exfalso
+--     apply Nat.not_lt_zero 1
+--     exact h
+--   · simp [Fin]
+--   done
+
+def ZMod.toInt (x : ZMod q) : Int := ZMod.cast x
+def ZMod.toFin (x : ZMod q) : Fin q := ZMod.cast x
+
+theorem ZMod.toInt_coe_eq (x : ZMod q) : ↑(x.toInt) = x := by
```

Comment:
this is `coe_toInt`

Reply:
That lemma doesn't seem to exist: https://github.com/search?q=repo%3Aleanprover-community%2Fmathlib4%20coe_toInt&type=code

Or are you suggesting to rename it to that?

Reply:
I was just suggesting a rename. Sorry for not being clear.